Automattic/simplenote-electron
# Add sorting bar to note list

### Fix

Adds a "Sort by" bar to the bottom of the notes list. See p2XJRt-2Ct-p2

Designs are at:
https://app.zeplin.io/project/5d795a3d8ed5261aa146e057/screen/5fd77f1846903987b1b40e75 (light mode)
https://app.zeplin.io/project/5d795a3d8ed5261aa146e057/screen/5fd77f196e43637c2cb2889f (dark mode)

Also shortens the search results bar to 44px to match the height of the sorting bar and the new toolbar-height.

Fixes #1625

### Test
1. Use the toggle to change the sort type and direction of the results.
2. Also try it with search filtering
3. Try in different browsers/OSes to make sure the dropdown styling is okay

### Screenshots

<img width="337" alt="Screen Shot 2021-02-02 at 11 26 14 AM" src="https://user-images.githubusercontent.com/52152/106651904-a0d0f980-6549-11eb-9d02-f7e7aa19520f.png">

<img width="331" alt="Screen Shot 2021-02-02 at 11 26 20 AM" src="https://user-images.githubusercontent.com/52152/106651913-a3cbea00-6549-11eb-9ee7-f116260fccf1.png">

<img width="320" alt="Screen Shot 2021-02-02 at 11 26 34 AM" src="https://user-images.githubusercontent.com/52152/106651929-a7f80780-6549-11eb-9958-7ba22dfad7c4.png">

<img width="256" alt="Screen Shot 2021-02-02 at 11 26 59 AM" src="https://user-images.githubusercontent.com/52152/106651937-ab8b8e80-6549-11eb-80a3-33ad69c408b8.png">

Note that in all browsers except Firefox, the dropdown styling is determined by the system theme, so it looks like this if you have a light system theme and the dark Simplenote theme:

<img width="317" alt="Screen Shot 2021-02-02 at 11 26 44 AM" src="https://user-images.githubusercontent.com/52152/106652316-205ec880-654a-11eb-90cd-835559bb909a.png">

In Firefox, we have to style the dropdown since otherwise it takes the styles from the parent, so I have added dark mode styles for Firefox that will be shown regardless of the system setting (in Simplenote dark theme):

<img width="315" alt="Screen Shot 2021-02-02 at 11 42 03 AM" src="https://user-images.githubusercontent.com/52152/106653934-4be2b280-654c-11eb-9b90-caa6df69120b.png">

The weird thing is that setting a background on the `select` seems to wipe out the padding and rounded edges (only in dark mode), but if we don't set it, we get this:

<img width="180" alt="Screen Shot 2021-02-02 at 12 05 48 PM" src="https://user-images.githubusercontent.com/52152/106656391-7bdf8500-654f-11eb-9c6a-b9d69a8a952f.png">

Firefox light theme changes the white background to a light gray, which matches default styling:

<img width="216" alt="Screen Shot 2021-02-02 at 11 33 15 AM" src="https://user-images.githubusercontent.com/52152/106652564-7b90bb00-654a-11eb-9447-900c5854aaca.png">

Search results bar shortened to match:

<img width="909" alt="Screen Shot 2021-02-02 at 12 02 51 PM" src="https://user-images.githubusercontent.com/52152/106656973-45eed080-6550-11eb-8955-5f4a590feb09.png">

<img width="873" alt="Screen Shot 2021-02-02 at 12 05 11 PM" src="https://user-images.githubusercontent.com/52152/106656992-4c7d4800-6550-11eb-92c9-626235b4eb67.png">


### Release

Added sort order bar to note list
